File: android/daggerdemo/app/src/main/java/com/example/fcl/daggerdemo/AppModule.java
package com.example.fcl.daggerdemo;

import android.content.Context;
import android.content.SharedPreferences;
import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private MyApp app;

    public AppModule(MyApp app) {
        this.app = app;
    }

    @Provides
    SharedPreferences provideSharedPreferences() {
        return app.getSharedPreferences("spfile", Context.MODE_PRIVATE);
    }

    @Provides
    MyApp provideApp() {
        return app;
    }

}

File: android/demo/app/src/main/java/com/example/fcl/dadademo/dagger/AppModule.kt
package com.example.fcl.dadademo.dagger

import com.example.fcl.kotlindemo.MyApp
import dagger.Module
import dagger.Provides

/**
 * Created by galio.fang on 18-8-7
 */

@Module
class AppModule(val myApp: MyApp) {

    @Provides
    fun providesApp() : MyApp {
        return myApp
    }

}